pkg/controller/master/addon/addon.go

```diff
@@ -282,14 +447,30 @@ func (h *Handler) ReconcileHelmChartOwners(_ string, _ string, obj runtime.Objec
 }
 
 func (h *Handler) removeHelmChart(aObj *harvesterv1.Addon) (*harvesterv1.Addon, error) {
-	a := aObj.DeepCopy()
-	err := h.checkAndDeleteChart(a)
+	// a := aObj.DeepCopy()
+	d, err := h.checkAndDeleteChart(aObj)
 	if err != nil {
-		return a, fmt.Errorf("error removing helm chart for addon %s: %v", a.Name, err)
+		return aObj, fmt.Errorf("error removing helm chart for addon %s: %v", aObj.Name, err)
 	}
 
-	a.Status.Status = ""
-	return h.addon.UpdateStatus(a)
+	// delete or deleting, make sure status is AddonDisabling
+	if d {
+		if aObj.Status.Status != harvesterv1.AddonDisabling {
+			a := aObj.DeepCopy()
+
+			logrus.Infof("addon %s is being disabled now, update status from %s to %s", a.Name, a.Status.Status, harvesterv1.AddonDisabling)
+			a.Status.Status = harvesterv1.AddonDisabling
+			return h.addon.UpdateStatus(a)
+		}
```

Comment:
we should add more checks here.. 

if status is already Addon.Disabling, then helmChart object will have a new `helm-delete` job submitted. we should query the job status for the same, and accordingly update the addon status to `AddonDisableFailed` or clear status if job was successful.